modules/giapi/src/main/scala/giapi/client/package.scala

```diff
@@ -157,42 +157,67 @@ package client {
         } yield i
       }
 
+    // taken from cats-effect 1.x
+    private def toIOFromRunAsync[F[_], A](f: F[A])(implicit F: Effect[F]): IO[A] =
+      IO.async { cb =>
+        F.runAsync(f)(r => IO(cb(r))).unsafeRunSync()
+      }
+
+    // Implementations of timeout suggested from cats-effect documentation
+    private def timeoutTo[F[_]: Effect, A](fa: F[A], after: FiniteDuration, fallback: F[A])
```

Comment:
This is taken from `cats-effect` but on `F[_]` rather than `IO`. It maybe better to use `IO` instead and convert to `F[_]` on the caller instead. I can't tell which one is better

Reply:
/cc @tpolecat

Reply:
So with cats-effect 1.0 you have `F <~> IO` given `Effect[F]` via `fa.toIO` and then `io.liftIO[F]` … so what I might do here is use an implicit class to add a `.toIO` method for any `F: Effect` (`.liftIO` already works) and then delegate the timeout stuff to the existing `IO` combinator. When cats-effect 1.0 comes out we can just delete the implicit class. Does that make sense?